File: rb/features.rb
# RuntimebuzzArticle SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module RuntimebuzzArticleFeatures
  def self.make_feature(name)
    case name
    when "base"
      RuntimebuzzArticleBaseFeature.new
    when "test"
      RuntimebuzzArticleTestFeature.new
    else
      RuntimebuzzArticleBaseFeature.new
    end
  end
end

File: py/features.py
# RuntimebuzzArticle SDK feature factory

from feature.base_feature import RuntimebuzzArticleBaseFeature
from feature.test_feature import RuntimebuzzArticleTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RuntimebuzzArticleBaseFeature(),
        "test": lambda: RuntimebuzzArticleTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
